osu.Game.Rulesets.Osu/Objects/Drawables/Connections/FollowPointConnection.cs

```diff
@@ -2,54 +2,50 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using System;
+using System.Diagnostics;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Pooling;
 using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Pooling;
 using osuTK;
 
 namespace osu.Game.Rulesets.Osu.Objects.Drawables.Connections
 {
     /// <summary>
     /// Visualises the <see cref="FollowPoint"/>s between two <see cref="DrawableOsuHitObject"/>s.
     /// </summary>
-    public class FollowPointConnection : PoolableDrawable
+    public class FollowPointConnection : PoolableDrawableWithLifetime<FollowPointLifetimeEntry>
     {
         // Todo: These shouldn't be constants
         public const int SPACING = 32;
         public const double PREEMPT = 800;
 
-        public FollowPointLifetimeEntry Entry;
         public DrawablePool<FollowPoint> Pool;
 
-        protected override void PrepareForUse()
+        protected override void OnApply(FollowPointLifetimeEntry entry)
         {
-            base.PrepareForUse();
+            base.OnApply(entry);
 
-            Entry.Invalidated += onEntryInvalidated;
-
-            refreshPoints();
+            entry.Invalidated += refreshPoints;
+            refreshPoints(entry);
         }
 
-        protected override void FreeAfterUse()
+        protected override void OnFree(FollowPointLifetimeEntry entry)
         {
-            base.FreeAfterUse();
-
-            Entry.Invalidated -= onEntryInvalidated;
+            base.OnFree(entry);
 
+            entry.Invalidated -= refreshPoints;
             // Return points to the pool.
             ClearInternal(false);
-
-            Entry = null;
         }
 
-        private void onEntryInvalidated() => Scheduler.AddOnce(refreshPoints);
```

Comment:
Can you explain why this schedule was removed so I don't have to dig for the reason?

Reply:
I missed it is scheduling. I'll back it because I'm not sure about effect.

Reply:
The scheduling was potentially unsafe (entry can be freed in that frame) so I added a guard.